sources/common/infoschema.go

```diff
@@ -86,6 +93,7 @@ func ProcessData(conv *internal.Conv, infoSchema InfoSchema) {
 		}
 		err := infoSchema.ProcessData(conv, srcTable, srcSchema, spTable, spCols, spSchema)
 		if err != nil {
+			writer.Flush()
```

Comment:
I think this writer.Flush() should come right after this if err != nil block.

Reply:
Doing that will also require a change in how we do Flush right now. We should probably add a similar option to pass a flush() function to the batchwriter, similar to how we pass a data sink. That, way the unit tests also won't require any major change.

Reply:
By passing it as a function to the batchwriter, I'm assuming you meant adding flush as a function param in conv to avoid passing writer as a param to infoschema. While I wouldn't want to add more to conv but exposing writer to infoschema is also not something that is desired.

Unrelated, 2 possible performance improvements unless they are already captured somewhere
1. I saw that we are flushing the data after entire data is processed. It should be better for performance if we flush data as soon as we have enough rows processed for write thresholds. This would make processing and writing somewhat parallel.
2. Each row and each table is processed sequentially currently, independent tables and rows should also be processed in parallel.

Reply:
Data is written in parallel currently via go routines and when write threshold reaches the system waits before processing further records.

Reply:
Writes do happen in parallel but looking at the code it seemed like the first write starts after entire processing. I might be missing something so we can discuss further. These were anyways not related to the PR. Just put it out here as a suggestion for future optimizations.